### PR TITLE
* Implement a PATCH endpoint for updating Name and qualificationName for Freelancer (add qualificationName as an optional column in Freelancers table).

### PR DESCRIPTION
**Pull request by ahmed-naeem2**

### CODE REVIEW

**Comment by KAJOiq** [comment]
Please resolve your Conflict.

File: Controllers/Mobile/v1/FreelancersController.cs [hunk 1]
```diff
@@ -97,5 +97,32 @@ public async Task<IActionResult> DeleteCertificationAsync([FromRoute] long certi
...
+        public async Task<IActionResult>UpdateFreelancer(long id, [FromBody]FreelancerUpdateDTO updateFreelancerDTO)
+        {
+            if(!ModelState.IsValid)
+
+                    return CustomBadRequest();
+
+            var stordFreelancer=await freelancerService.FindFreelancerById(id);
+
+            if (stordFreelancer == null)
+                return CustomBadRequest();
```
Comment (MohammedLilo) [inline Controllers/Mobile/v1/FreelancersController.cs:114]: if stored freelancer is null, you must return 404. as in other endpoints.

File: Controllers/Mobile/v1/FreelancersController.cs [hunk 2]
```diff
@@ -97,5 +97,32 @@ public async Task<IActionResult> DeleteCertificationAsync([FromRoute] long certi
...
+            if(!string.IsNullOrEmpty(updateFreelancerDTO.Qualification))
+            stordFreelancer.qualificationName=updateFreelancerDTO.Qualification;
```
Comment (MohammedLilo) [inline Controllers/Mobile/v1/FreelancersController.cs:116]: nigga you are assiging the same shit twice

Comment (MohammedLilo) [reply Controllers/Mobile/v1/FreelancersController.cs:116]: also, delete the if statement and perform the assignment directly.

File: Controllers/Web/v1/FreelancersController.cs
Comment (MohammedLilo) [inline Controllers/Web/v1/FreelancersController.cs:1]: same as mobile's version controller

File: Models/DTOs/FreelancerUpdateDTO.cs
Comment (MohammedLilo) [inline Models/DTOs/FreelancerUpdateDTO.cs:1]: TOO MANY WHITE SPACES

Comment (MohammedLilo) [inline Models/DTOs/FreelancerUpdateDTO.cs:1]: Use a name more descriptive than `FreelancerUpdateDTO`

**Review by MohammedLilo — COMMENTED** [review]
check out the comments, also don't spam white spaces everywhere, nigga

File: Controllers/Mobile/v1/FreelancersController.cs [hunk 3]
```diff
@@ -206,5 +206,28 @@ public async Task<IActionResult> DeleteEducationAsync([FromRoute] long education
             return NoContent();
         }
 
+
+
+        [Authorize(Roles = Constants.USER_TYPE_FREELANCER)]
+
+        [HttpPatch("Update/{id}")]
```
Comment (Yousif-Alsheikh) [inline Controllers/Mobile/v1/FreelancersController.cs:213]: update ?

File: Controllers/Mobile/v1/FreelancersController.cs [hunk 4]
```diff
@@ -206,5 +206,28 @@ public async Task<IActionResult> DeleteEducationAsync([FromRoute] long education
...
+
+        public async Task<IActionResult>UpdateQualificationAndName(long id, [FromBody]UpdateFreelancerQualificationAndName updateFreelancerQualificationAndName)
```
Comment (Yousif-Alsheikh) [inline Controllers/Mobile/v1/FreelancersController.cs:215]: edit the DTO name  "UpdateFreelancerQualificationAndName"

File: Controllers/Mobile/v1/FreelancersController.cs [hunk 5]
```diff
@@ -206,5 +206,28 @@ public async Task<IActionResult> DeleteEducationAsync([FromRoute] long education
...
+        {
+            var storedFreelancer=await freelancerService.FindFreelancerByIdAsync(id);
+
```
Comment (Yousif-Alsheikh) [inline Controllers/Mobile/v1/FreelancersController.cs:218]: check if the freelancer is the one edit his info.

File: Controllers/Web/v1/FreelancersController.cs [hunk 6]
```diff
@@ -201,5 +201,25 @@ public async Task<IActionResult> GetActivitiesAsync(long id)
             var responseDTO = activitiesService.FreelancerActivities(id);
             return Ok(CreateSuccessResponse(responseDTO));
         }
+        [Authorize(Roles = Constants.USER_TYPE_FREELANCER)]
+
+        [HttpPatch("Update/{id}")]
```
Comment (Yousif-Alsheikh) [inline Controllers/Web/v1/FreelancersController.cs:206]: "Update/{id}" ?

File: Controllers/Web/v1/FreelancersController.cs [hunk 7]
```diff
@@ -201,5 +201,25 @@ public async Task<IActionResult> GetActivitiesAsync(long id)
...
+
+        public async Task<IActionResult> UpdateQualificationAndName(long id, [FromBody] UpdateFreelancerQualificationAndName freelancerUpdateDTO)
```
Comment (Yousif-Alsheikh) [inline Controllers/Web/v1/FreelancersController.cs:208]: edit the DTO name "UpdateFreelancerQualificationAndName"

File: Controllers/Web/v1/FreelancersController.cs [hunk 8]
```diff
@@ -201,5 +201,25 @@ public async Task<IActionResult> GetActivitiesAsync(long id)
...
+        {
+            var storedFreelancer = await freelancerService.FindFreelancerByIdAsync(id);
+
+            if (storedFreelancer == null)
+                return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "The Freelancer Not Found"));
+
```
Comment (Yousif-Alsheikh) [inline Controllers/Web/v1/FreelancersController.cs:214]: check if the freelancer is the one edit his info .

File: Models/DTOs/UpdateFreelancerQualificationAndName.cs [hunk 9]
```diff
@@ -0,0 +1,11 @@
+﻿namespace AonFreelancing.Models.DTOs
+{
+    public class UpdateFreelancerQualificationAndName
+    {
+
+        public string Name { get; set; }    
+
+
+        public string? qualificationName { get; set; }
```
Comment (Yousif-Alsheikh) [inline Models/DTOs/UpdateFreelancerQualificationAndName.cs:9]: SpecializationName insted of qualificationName, and freelancer may have more than on specialization .

**Review by Yousif-Alsheikh — CHANGES_REQUESTED** [review]
please check your code again.

**Comment by KAJOiq** [comment]
@ahmed-naeem2 you should check out your friends comment, so you need to update your code then push it again.

**Pull request closed without merging**